### ISSUE TITLE
Instance methods have precedence over prefix operators

### ISSUE THREAD

**Issue by aost** [opal/opal#462]
```ruby
# Code
puts -5
puts (-5).abs
puts -(5.abs)
puts -5.abs

# Opal output
-5
5
-5
-5

# MRI output
-5
5
-5
5
```

**Comment by alexdowad** [comment]
The author of this grammar put an explicit precedence declaration on the rule which parses unary +/- followed by a number. It was given very, very low precedence to avoid a conflict with a rule which parses unary +/-, followed by a number, followed by the exponentiation operator.

**Comment by alexdowad** [comment]
...Which is understandable, because `**` actually has higher precedence than unary +/-. So `-1 ** 2` is `-(1 ** 2)`, not `(-1) ** 1`.

**Comment by alexdowad** [comment]
It looks like this was fixed somewhere since 2013, but nobody noticed!

**Comment by alexdowad** [comment]
This should be closed.

**Comment by elia** [comment]
Probably the issue is still there for method calls:

```
$ opal -e 'puts -5.abs' # => -5
$ ruby -e 'puts -5.abs' # => 5
```

**Comment by elia** [comment]
No, wait, it's wrong only if it's preceded by another call:

```
$ opal --sexp -e '-5.abs'                                                     ~
(:call, (:int, -5), :abs, (:arglist))
$ opal --sexp -e 'puts -5.abs'                                                ~
(:call, nil, :puts, (:argils, (:call, (:call, (:int, 5), :abs, (:arglist)), :-@, (:arglist))))
```

**Comment by alexdowad** [comment]
I've found that even MRI parses `-a.method` as `-(a.method)`. It seems to special-case numeric literals.

**Comment by alexdowad** [comment]
(Good find about method calls, by the way.)

**Comment by alexdowad** [comment]
My updated PR fixes this. I'm not versed in all the intricacies of your lexer/parser, so hopefully there won't be any adverse side effects! The fix *seems* right, but it would be nice if the original author also had a look.

**Comment by elia** [comment]
Looks good, and the test suite should a good protection against side effects 👍🏼

(Unfortunately the original author isn't active on the project anymore)

**Comment by alexdowad** [comment]
This is fixed and should be closed.

**Comment by elia** [comment]
Fixed by #1215
